### PR TITLE
feat: add pages in action forms

### PR DESCRIPTION
**Pull request by matthv**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forest_admin_agent/lib/forest_admin_agent/utils/schema/generator_action.rb [hunk 1]
```diff
@@ -73,6 +72,15 @@ def self.build_layout_schema(field)
               fields: field.fields.map { |f| build_layout_schema(f) }
             }
           end
+          if field.component == 'Page'
```
Comment (jbarreau) [inline packages/forest_admin_agent/lib/forest_admin_agent/utils/schema/generator_action.rb:75]: else if ?

File: packages/forest_admin_agent/spec/lib/forest_admin_agent/utils/schema/generator_action_spec.rb [hunk 2]
```diff
@@ -397,6 +397,80 @@ module Schema
...
+                schema: {
+                  actions: {
+                    'Charge credit card' => BaseAction.new(
+                      scope: Types::ActionScope::SINGLE
+                    )
+                  }
+                },
+                get_form: [
+                  ActionLayoutElement::PageElement.new(
+                    next_button_label: proc { |_context| 'Next' },
```
Comment (jbarreau) [inline packages/forest_admin_agent/spec/lib/forest_admin_agent/utils/schema/generator_action_spec.rb:412]: after speacking with Enky and Nico, we said theses fields are stirng only, no function allowed here right ?

File: packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/action/types/field_type.rb [hunk 3]
```diff
@@ -21,6 +21,8 @@ module FieldType
 
           JSON = 'Json'.freeze
 
+          LAYOUT = 'Layout'.freeze
```
[the remark this answers is not in the document]
Comment (matthv) [reply packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/action/types/field_type.rb:24]: yes 👀

File: packages/forest_admin_datasource_toolkit/lib/forest_admin_datasource_toolkit/components/actions/action_field_factory.rb [hunk 4]
```diff
@@ -22,6 +22,9 @@ def self.build_layout_element(field)
             return ActionLayoutElement::RowElement.new(**field) unless field[:fields].empty?
 
             nil
+          when 'Page'
+            # is it nullable if no elements ???
```
Comment (jbarreau) [inline packages/forest_admin_datasource_toolkit/lib/forest_admin_datasource_toolkit/components/actions/action_field_factory.rb:26]: Yes remove the page if it doesn't contains any element:
https://github.com/ForestAdmin/agent-nodejs/blob/main/packages/_example/src/forest/customizations/card.ts#L75

**Review by jbarreau — DISMISSED** [review]

File: packages/forest_admin_datasource_toolkit/lib/forest_admin_datasource_toolkit/components/actions/action_field_factory.rb (hunk 4 above)
Comment (matthv) [reply packages/forest_admin_datasource_toolkit/lib/forest_admin_datasource_toolkit/components/actions/action_field_factory.rb]: fixed :)

**Review by jbarreau — APPROVED** [review]

**Pull request merged**